cql3/Cql.g

```diff
@@ -1058,10 +1058,18 @@ dropIndexStatement returns [std::unique_ptr<drop_index_statement> expr]
     ;
 
 /**
-  * TRUNCATE <CF>;
+  * TRUNCATE <CF>
+  * [USING TIMEOUT]
   */
-truncateStatement returns [std::unique_ptr<truncate_statement> stmt]
-    : K_TRUNCATE (K_COLUMNFAMILY)? cf=columnFamilyName { $stmt = std::make_unique<truncate_statement>(cf); }
+truncateStatement returns [std::unique_ptr<raw::truncate_statement> stmt]
+    @init {
+        auto attrs = std::make_unique<cql3::attributes::raw>();
+    }
+    : K_TRUNCATE (K_COLUMNFAMILY)? cf=columnFamilyName
+      ( usingClause[attrs] )?
```

Comment:
There's no validation in this series so use of other `USING` attributes will succeed and they'll just be ignored.
As a follow up, I plan that after https://github.com/scylladb/scylladb/issues/11230 is done, the user could also use `TRUNCATE USING TIMESTAMP <timestamp>`, so it makes sense to set the stage for it and
maybe just add validation when preparing the truncate_statement internally from the raw statement.

Reply:
done in v2

Reply:
TRUNCATE USING TIMESTAMP requires truncate tombstones, since there could be data after the timestamp.

I don't see how @nyh's comment was addressed.